pkg/staking/stakingcontract/contract.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stakingcontract
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/sctx"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	MinimumStakeAmount = big.NewInt(1)
+
+	erc20ABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: get ABI for staking contract and replace it below
+	stakingABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: enable below mentioned topic for receiving receipts
+	//stakeUpdatedTopic = stakingABI.Events["StakeUpdated"].ID
+
+	ErrInsufficentStakeAmount = errors.New("insufficient stake amount")
+	ErrInsufficientFunds      = errors.New("insufficient token balance")
+	ErrNotImplemented         = errors.New("not implemented")
+	ErrGetStakeFailed         = errors.New("get stake failed")
+
+	depositStakeDescription = "Deposit Stake"
+)
+
+type Contract interface {
+	DepositStake(ctx context.Context, stakedAmount big.Int, overlay swarm.Address) error
+	GetStake(ctx context.Context, overlay swarm.Address) (big.Int, error)
+}
+
+type contract struct {
+	owner                  common.Address
+	stakingContractAddress common.Address
+	bzzTokenAddress        common.Address
+	transactionService     transaction.Service
+	overlayNonce           common.Hash
+}
+
+func New(
+	owner common.Address,
+	stakingContractAddress common.Address,
+	bzzTokenAddress common.Address,
+	transactionService transaction.Service,
+	nonce common.Hash,
+) Contract {
+	return &contract{
+		owner:                  owner,
+		stakingContractAddress: stakingContractAddress,
+		bzzTokenAddress:        bzzTokenAddress,
+		transactionService:     transactionService,
+		overlayNonce:           nonce,
+	}
+}
+
+func (s *contract) sendTransaction(ctx context.Context, callData []byte, desc string) (*types.Receipt, error) {
+	request := &transaction.TxRequest{
+		To:          &s.stakingContractAddress,
+		Data:        callData,
+		GasPrice:    sctx.GetGasPrice(ctx),
+		GasLimit:    1600000,
```

Comment:
We should also use the gas limit. Please check the open PR for gas limit and add same support here.